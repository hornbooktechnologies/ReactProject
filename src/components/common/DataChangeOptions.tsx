import React, { useRef } from 'react'
import DatePicker from 'react-datepicker'
import { useTranslation } from 'react-i18next'
import {
  DateformatConstant,
  ChartConstant,
} from '../../utils/AppConstants'
import { DataChangeOptionsModel } from '../../models/LinechartModel'
import { getSelectedValue } from '../../utils/AnnualTargetutils'
import 'react-datepicker/dist/react-datepicker.css'
import '../../assets/scss/datepicker.scss'

const DataChangeOptions = ({
  linechartData,
  onClickPrev,
  startDate,
  setStartDate,
  onClickNext,
  selectValue,
  customClass
}: DataChangeOptionsModel) => {
  const { i18n } = useTranslation();
  const datepickerRef = useRef(null)
  function handleClickDatepickerIcon() {
    const datepickerElement = datepickerRef.current
    datepickerElement.setFocus(true)
  }

  return (
    <>
      <div className="uk-margin-auto-left uk-flex uk-flex-middle uk-flex-wrap uk-width-1-1 uk-width-auto@s mt-15 top-m1">
        <div className="uk-flex uk-flex-middle uk-width-1-1 uk-width-auto@s">
          <button className="uk-button-outline" onClick={onClickPrev}>
            <i className="jera-arrow-left-2 color-white"></i>
          </button>
          <div className="uk-field-with-icon ml-10 m-w-200 uk-width-1-1 uk-width-auto@s mt-m14">
            <div className="form-control">
              <label className="uk-custom-label">
                <span>{i18n.t(getSelectedValue(selectValue))}</span>
              </label>
            </div>

            <div className="cal-inline">
              <p className="m-0 pr-4">{`${linechartData?.datelabel}`}</p>
            </div>
            <label className="mb-0">
              {selectValue === ChartConstant.ANNUAL_DATA ||
              selectValue === ChartConstant.CUMULATIVE_DATA ? (
                <DatePicker
                  selected={startDate}
                  format={DateformatConstant.YYYY}
                  ref={datepickerRef}
                  popperClassName={`${
                    customClass ? customClass : ''
                  } yearpicker`}
                  showYearPicker
                  onChange={(value) => setStartDate(value)}
                />
              ) : selectValue === ChartConstant.MONTHLY_DATA ? (
                <DatePicker
                  selected={startDate}
                  format={DateformatConstant.YYYY}
                  ref={datepickerRef}
                  popperClassName={`${
                    customClass ? customClass : ''
                  } monthpicker`}
                  showMonthYearPicker
                  onChange={(value) => setStartDate(value)}
                />
              ) : (
                <DatePicker
                  selected={startDate}
                  format={DateformatConstant.YYYY}
                  popperClassName={`${
                    selectValue === ChartConstant.WEEKLY_DATA
                      ? 'weekpicker'
                      : 'daypicker'
                  } ${customClass ? customClass : ''}`}
                  ref={datepickerRef}
                  onChange={(value) => setStartDate(value)}
                />
              )}

              <i
                className="jera-calendar-1 icn top-26"
                onClick={() => handleClickDatepickerIcon()}
              ></i>
            </label>
          </div>
          <button
            className="uk-button-outline ml-10"
            onClick={onClickNext}
            disabled={linechartData?.nextDisable}
          >
            <i className="jera-arrow-right-3 color-white"></i>
          </button>
        </div>
        {linechartData?.currentDisable === false ? (
          <button
            className="uk-button-outline filled uk-width-1-1 uk-width-auto@s uk-margin-small-left@s uk-margin-small-top uk-margin-remove-top@s"
            onClick={(e) => {
              setStartDate(new Date())
            }}
          >
            <i className="mr-5 font-20 jera-undo color-white"></i>
            {i18n.t('LABELS.RETURN')}
          </button>
        ) : null}
      </div>
    </>
  )
}

export default DataChangeOptions
