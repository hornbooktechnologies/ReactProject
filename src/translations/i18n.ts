import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
