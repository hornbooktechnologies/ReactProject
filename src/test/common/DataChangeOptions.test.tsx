import React from 'react'
import DataChangeOptions from '../../components/common/DataChangeOptions'
import renderer from 'react-test-renderer'
import i18n from '../../i18nForTests'
import { I18nextProvider } from 'react-i18next'
describe('Component: DataChangeOptions', () => {
  var Props  = {
    forecastHrs: 10,
    forecastpreviousHrs: 12,
    planHrs: 15,
    planpreviousHrs: 14,
    actual: true,
    suffix: 'oku',
  }
  var linechartData = {
    currentDisable: false,
    graphData:
      '{"Actua:[52.7,43.41,45.19,64.06,55.43,51.66,217.55,374.96,311.73,359.21,362.63,null],"Perio:["2021-04-01T00:00","2021-05-01T00:00","2021-06-01T00:00","2021-07-01T00:00","2021-08-01T00:00","2021-09-01T00:00","2021-10-01T00:00","2021-11-01T00:00","2021-12-01T00:00","2022-01-01T00:00","2022-02-01T00:00","2022-03-01T00:00"],"Planne:[15.31,18.49,49.32,68.43,-5.67,43.35,74.05,76.23,102.75,108.93,88.2,66.98],"Forecas:[null,null,null,null,null,null,null,null,null,null,null,438.43],"Estimate:[690.37,2078.44,206.54]}',
    indexFound: true,
    nextDisable: false,
    datelabel: '2021',
    Prefix: 'Â¥',
    Suffix: 'Oku',
    Suffix2: 'Oku',
  }
  var startDate = new Date();
  var selectValue = "AnnualData";
  var customClass = "date-picker-return";

  it('DataChangeOptions renders correctly', () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <DataChangeOptions
            {...Props}
            startDate={startDate}
            linechartData={linechartData}
            selectValue={selectValue}
            customClass={customClass}
          />
        </I18nextProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
