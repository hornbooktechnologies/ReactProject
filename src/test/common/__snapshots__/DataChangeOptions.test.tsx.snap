// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: DataChangeOptions DataChangeOptions renders correctly 1`] = `
<div
  className="uk-margin-auto-left uk-flex uk-flex-middle uk-flex-wrap uk-width-1-1 uk-width-auto@s mt-15 top-m1"
>
  <div
    className="uk-flex uk-flex-middle uk-width-1-1 uk-width-auto@s"
  >
    <button
      className="uk-button-outline"
    >
      <i
        className="jera-arrow-left-2 color-white"
      />
    </button>
    <div
      className="uk-field-with-icon ml-10 m-w-200 uk-width-1-1 uk-width-auto@s mt-m14"
    >
      <div
        className="form-control"
      >
        <label
          className="uk-custom-label"
        >
          <span>
            Year
          </span>
        </label>
      </div>
      <div
        className="cal-inline"
      >
        <p
          className="m-0 pr-4"
        >
          2021
        </p>
      </div>
      <label
        className="mb-0"
      >
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className=""
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={false}
              type="text"
              value="03/21/2022"
            />
          </div>
        </div>
        <i
          className="jera-calendar-1 icn top-26"
          onClick={[Function]}
        />
      </label>
    </div>
    <button
      className="uk-button-outline ml-10"
      disabled={false}
    >
      <i
        className="jera-arrow-right-3 color-white"
      />
    </button>
  </div>
  <button
    className="uk-button-outline filled uk-width-1-1 uk-width-auto@s uk-margin-small-left@s uk-margin-small-top uk-margin-remove-top@s"
    onClick={[Function]}
  >
    <i
      className="mr-5 font-20 jera-undo color-white"
    />
    LABELS.RETURN
  </button>
</div>
`;
